OrbitQt/ProfilingTargetDialog.cpp

```diff
@@ -0,0 +1,509 @@
+
+  process_proxy_model_.setSourceModel(&process_model_);
+  process_proxy_model_.setSortRole(Qt::EditRole);
+  process_proxy_model_.setFilterCaseSensitivity(Qt::CaseInsensitive);
+  ui_->processesTableView->setModel(&process_proxy_model_);
+  ui_->processesTableView->setSortingEnabled(true);
+  ui_->processesTableView->sortByColumn(static_cast<int>(ProcessItemModel::Column::kCpu),
+                                        Qt::DescendingOrder);
+
+  ui_->processesTableView->horizontalHeader()->resizeSection(
+      static_cast<int>(ProcessItemModel::Column::kPid), 60);
+  ui_->processesTableView->horizontalHeader()->resizeSection(
+      static_cast<int>(ProcessItemModel::Column::kCpu), 60);
+  ui_->processesTableView->horizontalHeader()->setSectionResizeMode(
+      static_cast<int>(ProcessItemModel::Column::kName), QHeaderView::Stretch);
+  ui_->processesTableView->verticalHeader()->setDefaultSectionSize(kProcessesRowHeight);
+  ui_->processesTableView->verticalHeader()->setVisible(false);
+
+  if (absl::GetFlag(FLAGS_local)) {
+    ui_->localFrame->setVisible(true);
+  }
+
+  QObject::connect(ui_->loadFromFileButton, &QPushButton::clicked, this,
+                   &ProfilingTargetDialog::SelectFile);
+  // This and the next connect makes the radiobuttons behave as if they were in a exclusive button
+  // group. If a user clicks on one of these and the it was not checked before, it is checked
+  // afterwards and the state machine transitions into the correct state, which "unchecks" the other
+  // buttons. If a user clicks on a radio button that is already checked, simply nothing happens,
+  // the button does not get unchecked.
+  QObject::connect(ui_->loadCaptureRadioButton, &QRadioButton::clicked, this, [this](bool checked) {
+    if (!checked) ui_->loadCaptureRadioButton->setChecked(true);
+  });
+  QObject::connect(ui_->localProfilingRadioButton, &QRadioButton::clicked, this,
+                   [this](bool checked) {
+                     if (!checked) ui_->localProfilingRadioButton->setChecked(true);
+                   });
+  QObject::connect(ui_->processesTableView->selectionModel(), &QItemSelectionModel::currentChanged,
+                   this, &ProfilingTargetDialog::ProcessSelectionChanged);
+  QObject::connect(ui_->processesTableView, &QTableView::doubleClicked, this, &QDialog::accept);
+  QObject::connect(ui_->processFilterLineEdit, &QLineEdit::textChanged, &process_proxy_model_,
+                   &QSortFilterProxyModel::setFilterFixedString);
+  QObject::connect(ui_->confirmButton, &QPushButton::clicked, this, &QDialog::accept);
+
+  if (!target_configuration_opt.has_value()) {
+    if (absl::GetFlag(FLAGS_local)) {
+      state_machine_.setInitialState(&state_local_);
+    } else if (ui_->stadiaWidget->IsActive()) {
+      state_machine_.setInitialState(&state_stadia_);
+    } else {
+      state_machine_.setInitialState(&state_file_);
+    }
+    ui_->stadiaWidget->Start(ssh_connection_artifacts);
+    return;
```

Comment:
So since if else are also not optimal, the best solution would probably be to split this part into an extra function and then it is an early return since it is closer to the start of the funktion?! Sounds like a good change to me

Reply:
Done. @beckerhe PTAL

Reply:
Looks good. I hope. It's really hard to navigate between commits and comments in GitHub. I'm wondering if I'm doing something wrong.